uamqp/_connection.py

```diff
@@ -201,12 +218,26 @@ def _send_frame(self, channel, frame, timeout=None, **kwargs):
         :param int timeout: An optional timeout value to wait until the socket is ready to send the frame.
         :rtype: None
         """
+        try:
+            raise self._error
+        except TypeError:
+            pass
+
         if self._can_write():
-            self._last_frame_sent_time = time.time()
-            if timeout:
-                with self._transport.block_with_timeout(timeout):
-                    self._transport.send_frame(channel, frame, **kwargs)
-            self._transport.send_frame(channel, frame, **kwargs)
+            try:
+                self._last_frame_sent_time = time.time()
+                if timeout:
+                    with self._transport.block_with_timeout(timeout):
+                        self._transport.send_frame(channel, frame, **kwargs)
+                self._transport.send_frame(channel, frame, **kwargs)
```

Comment:
hmm should line 232 be in an `else` statement`?
Otherwise it looks like, if `timeout` is specified, we're going to send twice?

Reply:
right, there should be an "else" statement.
it didn't run into error previously because we didn't use "timeout".